pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTMethodDeclaration.java

```diff
@@ -108,4 +111,25 @@ public ASTNameList getThrows() {
         }
         return null;
     }
+
+    @Override
+    public QualifiedName getQualifiedName() {
+        if (qualifiedName != null) {
+            return qualifiedName;
+        }
+
+        QualifiedName parent = this.getFirstParentOfType(ASTClassOrInterfaceDeclaration.class).getQualifiedName();
+
+        ASTFormalParameters params = this.getFirstDescendantOfType(ASTFormalParameters.class);
+        int numParams = params.jjtGetNumChildren();
+        String[] types = new String[numParams];
+
+        for (int i = 0; i < numParams; i++) {
+            types[i] = params.jjtGetChild(i).getFirstDescendantOfType(ASTType.class).getTypeImage();
```

Comment:
The same comment as for the ASTConstructorDeclaration applies here as well.

The implementation is basically the same for both nodes. Would it make sense, to move this method into `QualifiedName` and have here just a call, e.g.?

    if (qualifiedName = null) {
        qualifiedName = QualifiedName.determineOfConstructorOrMethod(this);
    }
    return qualifiedName;